app/frontend/src/features/connections/friends/useFriendList.test.tsx

```diff
@@ -0,0 +1,202 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { QueryClient, QueryClientProvider } from "react-query";
+import React from "react";
+import { service } from "../../../service";
+import useFriendList from "./useFriendList";
+import { getUser, listFriends } from "../../../test/serviceMockDefaults";
+
+const listFriendsMock = service.api.listFriends as jest.Mock;
+const getUserMock = service.user.getUser as jest.Mock;
+
+const wrapper = ({ children }: { children: React.ReactNode }) => {
+  const client = new QueryClient({
+    defaultOptions: {
+      queries: {
+        retry: false,
+      },
+    },
+  });
+
+  return <QueryClientProvider client={client}>{children}</QueryClientProvider>;
+};
+
+beforeAll(() => {
+  // Mock out console.error so the test output is less noisy when
+  // an error is intentionally thrown for negative tests
+  jest.spyOn(console, "error").mockReturnValue(undefined);
+});
+
+beforeEach(() => {
+  listFriendsMock.mockImplementation(listFriends);
+  getUserMock.mockImplementation(getUser);
+});
+
+describe("when the listFriends query is loading", () => {
+  it("returns loading with no errors and shouldn't try to load users", async () => {
+    const { result, waitForNextUpdate } = renderHook(() => useFriendList(), {
+      wrapper,
+    });
+
+    expect(result.current).toEqual({
+      isLoading: true,
+      isError: false,
+      errors: [],
+      friendQueries: [],
+    });
+    expect(getUserMock).not.toHaveBeenCalled();
+
+    await waitForNextUpdate();
```

Comment:
Why is waiting necessary after the assertions?

Reply:
This was there to get rid of that warning which says there's state updating outside of `act`, even though we don't care it about it in this case since we are interested in the loading state in this test rather than the eventual state when things get resolved successfully.